### PR TITLE
Allow at most one public constructor in @ScalarFunction class

### PR DESCRIPTION
**Pull request by haozhun**
I have talked to @martint about the design choice here.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/operator/TestAnnotationEngineForScalars.java [hunk 1]
```diff
@@ -67,8 +64,8 @@
     @Description("Simple scalar with single implementation based on class")
     public static class SingleImplementationScalarFunction
     {
-        @SqlType(DOUBLE)
-        public static double fun(@SqlType(DOUBLE) double v)
+        @SqlType(StandardTypes.DOUBLE)
```
Comment (wenleix) [inline presto-main/src/test/java/com/facebook/presto/operator/TestAnnotationEngineForScalars.java]: You are change from `String` to `Type` here. Will this fail? 

Same argument for `BIGINT`.

Comment (wenleix) [reply presto-main/src/test/java/com/facebook/presto/operator/TestAnnotationEngineForScalars.java]: nvm. It looks confusing when reviewing two commits together.

File: presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java [hunk 2]
```diff
@@ -451,7 +451,7 @@ public FunctionRegistry(TypeManager typeManager, BlockEncodingSerde blockEncodin
                 .scalars(SessionFunctions.class)
                 .scalars(StringFunctions.class)
                 .scalars(WordStemFunction.class)
-                .scalars(SplitToMapFunction.class)
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java]: typo in cmt msg "implentors"

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarFromAnnotationsParser.java [hunk 3]
```diff
@@ -52,7 +53,8 @@ private ScalarFromAnnotationsParser() {}
     {
         ImmutableList.Builder<SqlScalarFunction> builder = ImmutableList.builder();
         for (ScalarHeaderAndMethods methods : findScalarsInFunctionSetClass(clazz)) {
-            builder.add(parseParametricScalar(methods, FunctionsParserHelper.findConstructors(clazz)));
+            // Non-static function only makes sense in classes annotated @ScalarFunction.
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarFromAnnotationsParser.java]: nit: Maybe more explicit:

```
Only function definition classes (classes annotated with @ScalarFunction)
can define SQL functions via non-static java function. Function set classes
can only define SQL functions via static java functions.

Thus constructor in function set classes is used to define SQL functions. 
```

File: presto-main/src/test/java/com/facebook/presto/operator/scalar/TestScalarValidation.java [hunk 4]
```diff
@@ -370,13 +370,13 @@ public static long good2(
     @Test
     public void testValidTypeParametersForConstructors()
     {
-        extractScalars(ConstructorWithValidTypeParameters.class);
+        extractParametricScalar(ConstructorWithValidTypeParameters.class);
```
Comment (wenleix) [inline presto-main/src/test/java/com/facebook/presto/operator/scalar/TestScalarValidation.java]: I see. so `extractScalars` is for function set class and `extractParametricScalar` is for (parametric) function definition class.

File: presto-ml/src/main/java/com/facebook/presto/ml/MLFeaturesFunctions.java [hunk 5]
```diff
@@ -23,94 +23,189 @@
...
-import com.google.common.cache.CacheBuilder;
 import com.google.common.collect.ImmutableList;
-import com.google.common.hash.HashCode;
+
+import java.util.List;
 
 public final class MLFeaturesFunctions
 {
-    private static final Cache<HashCode, Model> MODEL_CACHE = CacheBuilder.newBuilder().maximumSize(5).build();
+    public static final List<Class<?>> ML_FEATURE_FUNCTIONS = ImmutableList.of(Features1.class, Features2.class, Features3.class, Features4.class, Features5.class, Features6.class, Features7.class, Features8.class, Features9.class, Features10.class);
```
Comment (wenleix) [inline presto-ml/src/main/java/com/facebook/presto/ml/MLFeaturesFunctions.java]: We should convert the implementation to use varargs. Filed issue: https://github.com/prestodb/presto/issues/10801

**Pull request merged**